### PR TITLE
Dotty Adjustments

### PR DESCRIPTION
**Pull request by cheeseng**
@liufengyun @bvenners 

Adjustments for latest Dotty.

Note that while AccumulationSpec now works, we have a regression in DefaultEqualitySpec (error during macro expansion).

Good news is the worrying compiler hanging problem now is fixed, though there're compiler error that's required further investigation.

### CODE REVIEW

**Comment by liufengyun** [comment]
The hang is fixed by https://github.com/lampepfl/dotty/pull/6163 and confirmed by the minimized test  https://github.com/lampepfl/dotty/pull/6166 .

The compilation error is due to this one: https://github.com/lampepfl/dotty/issues/6114

**Comment by liufengyun** [comment]
FYI: the regression in `DefaultEqualitySepc` is fixed in https://github.com/lampepfl/dotty/pull/6176, should compile tomorrow.

**Comment by cheeseng** [comment]
@liufengyun Cool!  I shall test it when it becomes available tomorrow, and remove the DefaultEqualitySpec in this PR once it is working.

Thanks for the info and update!

File: scalatest-test/src/test/scala/org/scalatest/AssertionsSpec.scala [hunk 1]
```diff
@@ -4969,25 +4971,25 @@ class AssertionsSpec extends AnyFunSpec {
...
       val e = intercept[TestCanceledException] {
-        assume(null == a, ". dude")
+        assume(null == s, ". dude")
       }
-      assert(e.message === Some(didNotEqual(null, 3) + ". dude"))
+      assert(e.message === Some(didNotEqual(null, "test") + ". dude"))
       assert(e.failedCodeFileName === (Some(fileName)))
       assert(e.failedCodeLineNumber === (Some(thisLineNumber - 4)))
     }
-    // SKIP-DOTTY-END
```
Comment (liufengyun) [inline scalatest-test/src/test/scala/org/scalatest/AssertionsSpec.scala]: Nice to remove the dotty skips 👍

**Comment by cheeseng** [comment]
@liufengyun fyi I made some progress on your CompileMacro implementation: 

https://github.com/scalatest/scalatest/pull/1559/commits/f1a6cbfc13a1c2e97bd174e7810d95976be1409d

**Comment by cheeseng** [comment]
@liufengyun A quick question, I wonder is there a way to differentiate a parse and type error?

Thanks!

**Comment by liufengyun** [comment]
@cheeseng Currently we don't have that, we are assuming library authors only interested in type errors.

Are there use cases for parsing errors? That seems to be testing the compiler instead of library.

**Comment by cheeseng** [comment]
@liufengyun Currently ScalaTest just give a different error message when it is failing, e.g. it is a parse/type error.  I guess it is fine if it is too much trouble, just a nice to have thing.  :)

**Pull request merged**